plexpy/webserve.py

```diff
@@ -1915,21 +1933,98 @@ def get_item_children(self, rating_key='', **kwargs):
 
     @cherrypy.expose
     @requireAuth()
-    def pms_image_proxy(self, img='', width='0', height='0', fallback=None, **kwargs):
+    def pms_image_proxy(self, img='', ratingkey=None, width='0', height='0', fallback=None, **kwargs):
+        """ Grabs the images from pms and saved them to disk """
 
-        pms_connect = pmsconnect.PmsConnect()
-        result = pms_connect.get_image(img, width, height, fallback)
+        if not img or ratingkey:
+            logger.debug('No image input received')
+            return
 
-        if result:
-            cherrypy.response.headers['Content-type'] = result[1]
-            return result[0]
-        else:
-            return None
+        if ratingkey and not img:
+            img = '/library/metadata/%s/thumb/1337' % ratingkey
+
+        i = img.rsplit('/', 1)
+        i += [width, height]
```

Comment:
Image url split but the timestamp is still there when you join it back together.